rust/worker/src/log/log.rs

```diff
@@ -0,0 +1,165 @@
+    client: LogServiceClient<tonic::transport::Channel>,
+}
+
+impl GrpcLog {
+    pub(crate) fn new(client: LogServiceClient<tonic::transport::Channel>) -> Self {
+        Self { client }
+    }
+}
+
+#[derive(Error, Debug)]
+pub(crate) enum GrpcLogError {
+    #[error("Failed to connect to log service")]
+    FailedToConnect(#[from] tonic::transport::Error),
+}
+
+impl ChromaError for GrpcLogError {
+    fn code(&self) -> ErrorCodes {
+        match self {
+            GrpcLogError::FailedToConnect(_) => ErrorCodes::Internal,
+        }
+    }
+}
+
+#[async_trait]
+impl Configurable for GrpcLog {
+    async fn try_from_config(worker_config: &WorkerConfig) -> Result<Self, Box<dyn ChromaError>> {
+        match &worker_config.log {
+            LogConfig::Grpc(my_config) => {
+                let host = &my_config.host;
+                let port = &my_config.port;
+                println!("Connecting to log service at {}:{}", host, port);
+                let connection_string = format!("http://{}:{}", host, port);
+                let client = LogServiceClient::connect(connection_string).await;
+                match client {
+                    Ok(client) => {
+                        return Ok(GrpcLog::new(client));
+                    }
+                    Err(e) => {
+                        return Err(Box::new(GrpcLogError::FailedToConnect(e)));
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl Log for GrpcLog {
+    async fn read(
+        &mut self,
+        collection_id: String,
+        index: i64,
+        batch_size: i32,
+    ) -> Vec<Box<EmbeddingRecord>> {
+        let request = self.client.pull_logs(chroma_proto::PullLogsRequest {
+            collection_id: collection_id,
+            start_from_id: index,
+            batch_size: batch_size,
+        });
+        let response = request.await;
+        match response {
+            Ok(response) => {
+                let logs = response.into_inner().records;
+                let mut result = Vec::new();
+                for log in logs {
+                    let embedding_record: Result<EmbeddingRecord, EmbeddingRecordConversionError> =
+                        log.try_into();
+                    match embedding_record {
+                        Ok(embedding_record) => {
+                            result.push(Box::new(embedding_record));
+                        }
+                        Err(err) => {
+                            println!("Failed to convert log to embedding record: {}", err);
+                            // TODO: error handling
+                            return Vec::new();
+                        }
+                    }
+                }
+                result
+            }
+            Err(e) => {
+                println!("Failed to pull logs: {}", e);
+                Vec::new()
+            }
+        }
+    }
+
+    async fn get_collections(&mut self) -> Vec<CollectionInfo> {
```

Comment:
Why is this named get_collections? I find this name confusing and not aligned with the service